### ISSUE TITLE
Wrong debug output with proxy command

### ISSUE THREAD

**Issue by imoverclocked** [net-ssh/net-ssh#512]
### Expected behavior

When using a ProxyCommand or similar config the debug output should resemble:

```
D, [2017-04-04T20:50:53.786036 #41101] DEBUG -- net.ssh.transport.session[3fdb40d9d8d8]: establishing connection to proxy command
```

### Actual behavior

Currently, the behavior in debug mode prints a line:

```
D, [2017-04-04T20:50:53.786036 #41101] DEBUG -- net.ssh.transport.session[3fdb40d9d8d8]: establishing connection to some.host:22
```

This disregards proxy configurations where we are not actually connecting to the host:port and can throw debugging off.

https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/transport/session.rb#L64

### Example App

```ruby
gem 'net-ssh', '= 4.0.0.beta3'
require 'net/ssh'
puts Net::SSH::Version::CURRENT

@host = 'localhost'
@user = ENV['USER']
Net::SSH.start(@host, @user, config: 'proxy_command_config') do |ssh|
  puts ssh.exec!('echo "hello"')
end
```

proxy_command_config:
```
Host *
  ProxyCommand nc %h 8080
```

**Comment by mfazekas** [comment]
Thanks for the detailed issue, are you interseted in making a PR solving it?

**Comment by imoverclocked** [comment]
I can't at the moment.

**Comment by j1wilmot** [comment]
I believe this was resolved in #530 and can be closed.
